ibis/expr/types/relations.py

```diff
@@ -682,13 +698,18 @@ def __getitem__(self, what: str | int | slice | Sequence[str | int]):
         """
         from ibis.expr.types.logical import BooleanValue
 
-        if isinstance(what, str):
-            return ops.Field(self.op(), what).to_expr()
-        elif isinstance(what, int):
-            return ops.Field(self.op(), self.columns[what]).to_expr()
-        elif isinstance(what, slice):
-            limit, offset = util.slice_to_limit_offset(what, self.count())
-            return self.limit(limit, offset=offset)
+        try:
+            if isinstance(what, str):
+                return ops.Field(self.op(), what).to_expr()
+            elif isinstance(what, int):
+                return ops.Field(self.op(), self.columns[what]).to_expr()
+            elif isinstance(what, slice):
+                limit, offset = util.slice_to_limit_offset(what, self.count())
+                return self.limit(limit, offset=offset)
+        except com.FieldsNotFoundError:
+            # the raised FieldsNotFoundError contains the Op, but we want
+            # to only expose the Expr.
```

Comment:
Let's avoid storing unused state on exceptions. This can have unintended consequences for object lifetimes due to exceptions keeping stack frames around for tracebacks.

Please remove any unused fields from your new exception types.

Reply:
Oh wow, I'd never thought about how that might effect lifetimes, that's really interesting. Sounds good, I am removing them. I originally had them because in an earlier version I did actually need to keep track of them, but now I don't need that, so I'll stop storing them.